enhancements/sig-architecture/11-cluster-deletion/README.md

```diff
@@ -0,0 +1,122 @@
+   ```
+      /registrion
+      -- controller
+      -- forceDeletionGracePeriodSeconds: 600
+      -- resources-to-monitor-for-deletion: resource.version.group,resource.version.group.com...
+   ```
+   
+   * `forceDeletionGracePeriodSeconds` specifies the duration seconds of the cluster is unavailable after the cluster is deleting. The `managedClusterAddons` and `manifestWorks` which are not deleted because of the finalizers will be force deleted after the duration seconds. The default is `600s`.
+
+   * `resources-to-monitor-for-deletion` specifies a list of resources GVR. registration-controller will not delete anything until all resources defined here are deleted by other controller. The common style of string which may be either `resource.version.group` or `resource.version.group.com`. 
+
+4. Add a managedCluster Deletion controller in registration-controller to handle the resource cleanup. The deletion process like this:
+   
+   1. The cluster admin deletes the `managedCluster`. The `managedCluster` is under deleting since there is cleanup finalizer on it.
+   2. `deletion-controller` begins to monitor predeleted resources defined in `resources-to-monitor-for-deletion` at first, do not delete anything until all resources defined are cleaned up by other controller.
+   3. `deletion-controller` begins to delete all `managedClusterAddons`.
+   4. `deletion-controller` begins to delete all `manifestWorks` without the label `cluster.open-cluster-management.io/delete-by-other` after all `managedClusterAddons` are deleted.
+   5. `deletion-controller` begins to delete `roleBindings` of registration-agent and work-agent.
+   6. `deletion-controller` begins to delete cluster namespace without the label `cluster.open-cluster-management.io/delete-by-other` after all `manifestWorks` are deleted.
+   7. `deletion-controller` begins to force delete the `managedClusterAddons`,`manifestWorks` and cluster namespace without the label `cluster.open-cluster-management.io/delete-by-other` after the `forceDeletionGracePeriodSeconds`.
+   8. `deletion-controller` remove the finalizer `cluster.open-cluster-management.io/api-resource-cleanup` on the `managedCluster`.
+
+5. We only guarantee that all `managedClusterAddons`, `manifestWorks` and `rbac` in the cluster namespace are deleted. The user should clean up the remaining resources in the cluster namespace manually if the namespace is failed to be deleted because of the contents remaining.
+
+
+#### manifestworkvalidators-webhook 
+
+manifestworkvalidators webhook will deny the creation of all `managedClusterAddons` and `manifestWorks` except addon `manifestWorks` when the `managedCluster` is deleting. Because `registration-controller` starts the resources' deletion process in cluster namespace when the `managedCluster` is deleting. But the pre-delete addon `manifestWork` is an exception since it is created when the `managedClusterAddon` is deleted.
```

Comment:
these type of manifestworks may need another label. Also what if a cluster is not found?

Reply:
how about add a label `cluster.open-cluster-management.io/created-by-addon` for the addon manifestWorks created by add-framework ?

added the case that webhook will deny the creation of all `managedClusterAddons` and `manifestWorks` when there is no `managedCluster` with the same name as the namespace.

Reply:
1. manifestworkvalidators webhook will deny the creation of all `managedClusterAddons` when the `managedCluster` is deleting.

2. manifestworkvalidators webhook will deny the creation of all `managedClusterAddons` and `manifestWorks` when there is no `managedCluster` with the same name as the namespace.